src/common/simple_spin.cc

```diff
@@ -31,9 +28,15 @@ void simple_spin_lock(simple_spinlock_t *lock)
 	return;
     }
     // delay
-    for (int i = 0; i < 100000; i++) {
-      *foo = (*foo * 33) + 17;
-    }
+#if defined(__i386_) || defined(__x86_64__)
+    asm volatile("pause");
+#elif defined(__arm__) || defined(__aarch64__)
+    asm volatile("yield");
+#elif defined(__ppc64__)
+    asm volatile("or 27,27,27");
+#else
+    static_assert(false,"Unknown architecture");
```

Comment:
or just `#error "Unknown architecture"`. which can be handled by the preprocessor earlier. and you might want to add a space before `"`.